### DISCUSSION TITLE
"Potential NSFW content was detected" from horde request with NSFW enabled

### DISCUSSION THREAD

**Discussion by axemaster** [sdwebui-w-horde/sd-webui-stable-horde-worker#86] [General]
I received this message on the console for an incoming request from the horde. My question is, was this returned to the horde and then onto the requestor? Why wasn't it honored? Although informative to me, it's not actionable by me. NSFW is enabled so this surprised me. I don't know where it came from. I have no NSFW content blocks enabled in A1111. I'll put the prompt requested below. It does not seem offensive.

`Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.`

The request. I don't know what the 4 character non-english part is. Google didn't either.

```
a humanized standing пшкфgirafe wearing clothes with exaggerated long neck
Steps: 30, Sampler: Euler a Karras, CFG scale: 7.5, Seed: 1907009863, Size: 512x512, Model hash: b4391b7978, Model: nsfw/deliberate_v2, Seed resize from: -1x-1, Denoising strength: 0.75
```

**Comment by axemaster** [comment]
> The request. I don't know what the 4 character non-english part is. Google didn't either

I've been informed "пшкф" is unicode that gets transliterated to NSFW. So probably that. Someone is trying to circumvent the filter while submitting a barely NSFW prompt at low rez that would breeze through anyway. Likely a new user.
 
But that's a lame trigger for a non-extant filter. Also still don't understand where this filter is.

**Comment by MaikoTan** [comment]
The end user could enable the NSFW filter by their own, like [`Stable UI`](https://aqualxx.github.io/stable-ui/) has the NSFW option.
![image](https://user-images.githubusercontent.com/19927330/225556699-07f56468-13ff-474c-bb4c-af4b2dbf60bf.png)

When this option was enabled in the incoming request, the extension would enable the NSFW filter no matter the SD-WebUI side have set the NSFW or not.

So this is not the fault of the extension itself or even yours, the extension just following the request from end users.

**Reply by axemaster** [reply]
Hmm horde should probably not send these to NSFW workers. At least prioritize such workers  way down.

**Reply by db0** [reply]
NSFW workers can generate SFW and NSFW requests. Only SFW workers are excluded from NSFW requests
